src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
+              case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+            }
+          }
+          val returnObject = JsonNodeFactory.instance.objectNode()
+          val arrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(validTitles)
+          returnObject.set("lang", arrayNode)
+          (returnObject, warnings, csvwPropertyType)
+        }
+        case o:ObjectNode => {
+          val valueCopy = o.deepCopy()
+          val fieldsAndValues = Array.from(valueCopy.fields.asScala)
+          for(fieldsAndValue <- fieldsAndValues) {
+            val elementKey = fieldsAndValue.getKey
+            val matcher = PropertyChecker.Bcp47LanguagetagRegExp.pattern.matcher(elementKey)
+            if(matcher.matches()) {
+              var validTitles = Array[String]()
+              fieldsAndValue.getValue match {
+                case s:TextNode => validTitles = validTitles :+ s.asText()
+                case a:ArrayNode => {
+                  val titles = Array.from(a.elements().asScala)
+                  for(title <- titles) {
+                    title match {
+                      case s:TextNode => validTitles = validTitles :+ s.asText()
+                      case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+                    }
+                  }
+                }
+                case _ => warnings = warnings :+ PropertyChecker.invalidValueWarning
+              }
+              val validTitlesArrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(validTitles)
+              valueCopy.set(elementKey, validTitlesArrayNode)
+            } else {
+              valueCopy.remove(elementKey)
+              warnings = warnings :+ "invalid_language"
+            }
+          }
+          (valueCopy, warnings, csvwPropertyType)
+        }
+        case _ => (NullNode.getInstance(), Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def nameProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode => {
+          val matcher = PropertyChecker.NameRegExp.pattern.matcher(s.asText())
+          if (matcher.matches()) {
+            (s, Array[String](), PropertyType.Column)
+          } else {
+            (NullNode.instance, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Column)
+          }
+        }
+        case _ => (NullNode.instance, Array[String](PropertyChecker.invalidValueWarning), PropertyType.Column)
+      }
+    }
+  }
+
+  def encodingProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode if PropertyCheckerConstants.ValidEncodings.contains(s.asText()) => {
```

Comment:
Again, should we be doing case-insensitive matching here? What does the CSVW spec say?

Reply:
As mentioned earlier, going with the Ruby implementation here